src/translate.py

```diff
@@ -3450,9 +3451,13 @@ def reg_at(index: int) -> List[Register]:
         or mnemonic in arch.instrs_no_dest
     ):
         return []
-    if mnemonic == "jal":
-        fn_target = instr.args[0]
-        if isinstance(fn_target, AsmGlobalSymbol):
+    if mnemonic in arch.instrs_store_update:
+        return reg_at(1)
+    if mnemonic in arch.instrs_load_update:
+        return reg_at(0) + reg_at(1)
+    if mnemonic in arch.instrs_fn_call:
+        if instr.args and isinstance(instr.args[0], AsmGlobalSymbol):
+            fn_target = instr.args[0]
             if global_info.is_function_known_void(fn_target.symbol_name):
                 return []
     if mnemonic in arch.instrs_fn_call:
```

Comment:
```suggestion
```

Reply:
Oh no, I didn't mean to remove the whole block, I just meant the two identical if checks
```
    if mnemonic in arch.instrs_fn_call:
        if instr.args and isinstance(instr.args[0], AsmGlobalSymbol):
            ...
        ...
    if mnemonic in arch.instrs_fn_call:
        ...
```
could be merged together.

Reply:
oh, hah, I thought it was because of the `return []` at the end of the function (but turns out you still need the check because of that second check)